### ISSUE TITLE
React state resets after componentWilllMount actions are triggered, without updating the DOM.

### ISSUE THREAD

**Issue by williamgranli** [react-boilerplate/react-boilerplate#1770]
Okay, so today I discovered a bug which affects all routes of my application and because of how it behaves it's affects the user in certain cases. 

When I reload a page, the functions in componentWillMount are fired away and finish correctly (async functions using sagas). What happens after is what I can't explain: the redux state resets back to the initial state, wipes the redux console (chrome addon) WITHOUT updating the DOM.

This happens in a split second, but you can see the redux console displaying the action from componentWillMount as triggered, but half a second later the list of actions are wiped and it just says "@@INIT" without any actions below, even though the DOM has been updated correctly.

Does anyone have any idea what the reason for this could be, or where I should start looking? It's quite tricky to find the source since it affects the whole application... We only have this problem when refreshing the page, by the way (pressing F5 or hitting the refresh button) and not when switching routes by pressing buttons on the page (redux-router push).

Let me know if you want me to post some code that might be relevant to the issue.

Thanks!

**Comment by gihrig** [comment]
> We only have this problem when refreshing the page, by the way (pressing F5 or hitting the refresh button) and not when switching routes by pressing buttons on the page 

Isn't that the expected behavior?

Sounds like you are looking for a persisted storage feature for application state. I'm not aware of any such feature in react-boilerplate. Did I miss something?

**Comment by williamgranli** [comment]
@gihrig I *think* you misunderstood what I wrote, let me explain in a clearer way.

(1). After I hit F5, the page reloads and the state resets, which is expected behavior.
(2). What happens after is that the functions in componentWillMount run. Also expected behavior.
(3). After this, the "bug" occurs. The redux state is wiped, all previous actions that have been dispatched are wiped from the redux console WITHOUT updating the dom.

So basically it seems that something goes wrong when the componentWillMount functions are ran. This affects all my pages and I've also tried to switch out the functions in componentWillMount for new ones, but the bug is still there.

Hopefully that made it more clear, @gihrig 

Thanks!

**Comment by gihrig** [comment]
@williamgranli OK, I think I understand, but to be sure, let me add to your 1, 2, 3 steps above.

1. ...
2. functions in componentWillMount run `>>>` actions are dispatched, updating redux state.
3. `>>>` Components complete mounting lifecycle, redux state is reset to "initial state", no DOM updates occur.

Assuming the above is correct, I would guess that something is dispatching an action with an empty or invalid payload after componentWillMount (in render method?).  The result being that your reducers are resetting to default state.

Can you confirm this by placing `console.log` or `debugger` statements in your reducers?

Not sure [this](https://busypeoples.github.io/post/react-component-lifecycle/) will help, it's a bit old, but should still be valid and does not take redux into consideration. `componentWillMount` and `render` detail:

>componentWillMount is called before the render method is executed. It is important to note that setting the state in this phase will not trigger a re-rendering.
>
>The render method returns the needed component markup, which can be a single child component or null or false (in case you don't want any rendering).

Could it be that `render` contains something that is returning false, null, or undefined?
